Sources/PropertyEditor.swift

```diff
@@ -0,0 +1,698 @@
+	/// `validations` signal.
+	///
+	/// The validation results of `inner` would be propagated to the created
+	/// `PropertyEditor`.
+	///
+	/// - parameters:
+	///   - inner: The inner property to wrap.
+	///   - transform: The value transform for the presentation.
+	///   - errorTransform: The error transform for the presentation.
+	///   - body: The closure to invoke for any proposed value to `self`.
+	public convenience init<T: _PropertyEditorProtocol>(
+		_ inner: T,
+		transform: @escaping (T.Value) -> Value,
+		errorTransform: @escaping (T.ValidationError) -> ValidationError,
+		_ body: @escaping (Value) -> Result<T.Value, ValidationError>
+	) {
+		self.init(inner, transform: transform, validator: body, validationSetup: { validations in
+			validations <~ inner.property.result.signal
+				.map { $0.map(transform, errorTransform, validator: body) }
+		})
+	}
+
+	/// Create an `PropertyEditor` that presents an editing interface of `inner`
+	/// in another value type, using the given forward transform and the given
+	/// failable reverse transform.
+	///
+	/// If `success` is returned by `body`, the associated value would be
+	/// persisted to `inner`. Otherwise, the failure would be emitted by the
+	/// `validations` signal.
+	///
+	/// - parameters:
+	///   - inner: The inner property to wrap.
+	///   - transform: The value transform for the presentation.
+	///   - body: The closure to invoke for any proposed value to `self`.
+	public convenience init<T: _PropertyEditorProtocol>(
+		_ inner: T,
+		transform: @escaping (T.Value) -> Value,
+		_ body: @escaping (Value) -> Result<T.Value, ValidationError>
+	) where T.ValidationError == NoError {
+		self.init(inner, transform: transform, validator: body, validationSetup: nil)
+	}
+
+	private init<T: _PropertyEditorProtocol>(
+		_ inner: T,
+		transform: @escaping (T.Value) -> Value,
+		validator: @escaping (Value) -> Result<T.Value, ValidationError>,
+		validationSetup: ((MutableProperty<ValidationResult<Value, ValidationError>>) -> Void)?
+	) {
+		let _validations: MutableProperty<ValidationResult<Value, ValidationError>> = inner.property.rootLock.lock {
+			let property = MutableProperty(ValidationResult(inner.property.committed.value, transform: transform, validator: validator))
+			validationSetup?(property)
+			return property
+		}
+
+		self.result = Property(capturing: _validations)
+		self.lifetimeToken = Lifetime.Token()
+		self.lifetime = Lifetime(lifetimeToken)
+		self.committed = inner.property.committed.map(transform)
+		self.rootLock = inner.property.rootLock
+
+
+		action = { input in
+			switch validator(input) {
+			case let .success(innerResult):
+				inner.property.action(innerResult)
+
+			case let .failure(error):
+				_validations.value = .failure(input, error)
+			}
+		}
+	}
+
+	/// Attempt to edit the property with the given value.
+	///
+	/// - parameters:
+	///   - newValue: The proposed value.
+	public func attemptSet(_ newValue: Value) {
```

Comment:
Why let values be set in this direction? Bidirectionality adds a LOT of complexity, and I can only think of 1 instance where I've ever wanted something  like this.